#!/usr/bin/env python
PACKAGE = "erdc_cloud_segmenter"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("use_roi_filter",           bool_t, 0, "",   True)
gen.add("use_ground_segmenter",     bool_t, 0, "",   True)
gen.add("use_nonground_segmenter",  bool_t, 0, "",   True)

gen.add("sor_filter",               bool_t,   0, "Statistical Outlier Filter",           True)
gen.add("sor_meanK",                int_t,    0, "K neighbors for SOR filter",             10,    1,   100)
gen.add("sor_stddevMulThresh",      double_t,  0, "sor_stddevMulThresh",                  0.1,    0,   3.0)

exit(gen.generate(PACKAGE, "segmenter_node", "segmenter"))
exit(gen.generate(PACKAGE, "segmenter_nodelet", "segmenter"))
